src/scenes/App/components/TermsAndConditions/index.jsx

```diff
@@ -0,0 +1,274 @@
+import { FormGroup, FormControlLabel } from 'material-ui/Form';
+import Checkbox from 'material-ui/Checkbox';
+import Button from 'material-ui/Button';
+
+import AppConfig from '../../../../config/app';
+import styles from './styles';
+
+const TNC_ACCEPTED = 'tncAccepted';
+
+@withStyles(styles, { withTheme: true })
+export default class TermsAndConditions extends React.PureComponent {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    locale: PropTypes.string.isRequired,
+    langHandler: PropTypes.func,
+  };
+
+  static defaultProps = {
+    langHandler: undefined,
+  };
+
+  state = {
+    checkboxChecked: false,
+  };
+
+  render() {
+    const { classes, locale } = this.props;
+    const { checkboxChecked } = this.state;
+
+    let tncAccepted = localStorage.getItem(TNC_ACCEPTED);
+    tncAccepted = _.isUndefined(tncAccepted) || _.isNull(tncAccepted) ? false : tncAccepted;
+
+    return (
+      <div
+        className={classes.tncBg}
+        style={
+          {
+            opacity: tncAccepted ? 0 : 1,
+            display: tncAccepted ? 'none' : 'block',
+          }
+        }
+      >
+        <Grid container>
+          <Grid item xs>
+            <Button color="primary" variant="raised" className={classes.langBtn} onClick={this.props.langHandler}>
+              <FormattedMessage id="language.select" defaultMessage="中文" />
+            </Button>
+          </Grid>
+          <Grid item xs={8}>
+            <Paper className={classes.tncWrapper}>
+              {this.renderTitle()}
+              {this.renderGeneral()}
+              {this.renderDefinitions()}
+              {this.renderSubordination()}
+              {this.renderParticipate()}
+              {this.renderRepresentations()}
+              {this.renderAwards()}
+              {this.renderPowers()}
+              {this.renderResponsibilities()}
+              {this.renderIp()}
+              {this.renderSupport()}
+              {this.renderMisc()}
+            </Paper>
+            <FormGroup row className={classes.formGroup}>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    checked={checkboxChecked}
+                    value="checkboxChecked"
+                    onChange={this.handleCheckedChange('checkboxChecked')}
+                  />
+                }
+                label={<FormattedMessage id="tnc.acceptCheckboxText" defaultMessage="I accept the Terms and Conditions" />}
+              />
+              <Button
+                type="submit"
+                color="primary"
+                variant="raised"
+                disabled={!checkboxChecked}
+                className={classes.acceptButton}
+                onClick={this.onAcceptButtonClick}
+              >
+                <FormattedMessage id="tnc.accept" defaultMessage="Accept" />
+              </Button>
+            </FormGroup>
+          </Grid>
+          <Grid item xs></Grid>
```

Comment:
What is this doing?

Reply:
its giving equal space on the left and right sides. in this code, having `<Grid item xs={8}>`, with two `<Grid item xs>` gives equal space for the two without the number assigned. so technically those two without numbers have it assigned as width 2